projects/ngx-sub-form/src/lib/ngx-sub-form-utils.ts

```diff
@@ -32,6 +34,37 @@ export type FormErrors<FormInterface> = null | Partial<
   }
 >;
 
+type Omit<T, K> = Pick<T, Exclude<keyof T, K>>; // can be removed when upgrading ts version
+type UntypedControlBase<TControl extends AbstractControl | FormArray> = Omit<
+  TControl,
+  'value' | 'valueChanges' | 'setValue' | 'patchValue'
+>;
+
+export type TypedAbstractControl<TValue extends any> = {
+  value: TValue;
+  valueChanges: Observable<TValue>;
+  setValue(
+    value: TValue,
+    options?: {
+      onlySelf?: boolean;
+      emitEvent?: boolean;
+      emitModelToViewChange?: boolean;
+      emitViewToModelChange?: boolean;
+    },
+  ): void;
+  patchValue(
+    value: Partial<TValue>,
+    options?: {
+      onlySelf?: boolean;
+      emitEvent?: boolean;
+      emitModelToViewChange?: boolean;
+      emitViewToModelChange?: boolean;
+    },
+  ): void;
+} & UntypedControlBase<AbstractControl>;
+
+export type TypedFormArray<TValue extends any> = TypedAbstractControl<TValue> & UntypedControlBase<FormArray>;
```

Comment:
Here you should not use `TypedAbstractControl` because the `options` property differs between `FormArray` and `AbstractControl`. We'll need to different definitions

Reply:
I'll have a look and push an update, likely over the WE

Reply:
There is now a `TypedControlBase` and both `TypedAbstractControl` and `TypedFormArray` use that. In addition the `UntypedControlBase` is now also dependent on `TypedControlBase` so there is never a miss match in which props to exclude.